GoogleChromeLabs/ps-analysis-tool
# Feature: Add topics taxonomy tree

## Description
This PR adds the topics taxonomy hierarchical table into a collapsible tree, it renders V1 and V2 of topic taxonomy with a search bar allowing searching of multiple leveled topics.


## Relevant Technical Choices
- D3JS is used to create a tree that handles user click events for adding and removing nodes.


## Testing Instructions
- Open the PSAT panel, and move to the topics component.
- Select the Taxonomy V2/V1 tab, a tree will load up.
- Click on a circular node will open the child node, whereas the rectangle as the leaf node.
- Try searching a topic and select it, it should toggle the node.


## Additional Information:



## Screenshot/Screencast

https://github.com/user-attachments/assets/9b283cb4-6911-4210-b355-f9081ae59b0e







---

## Checklist



- ~This code is covered by unit tests to verify that it works as intended.~



Fixes #783
